velox/exec/tests/AggregationFuzzer.cpp

```diff
@@ -794,7 +849,94 @@ void AggregationFuzzer::testPlan(
   }
 }
 
-void AggregationFuzzer::verify(
+std::string makeDuckWindowSql(
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<std::string>& sortingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<std::string>& inputs) {
+  std::stringstream sql;
+  sql << "SELECT " << folly::join(", ", inputs) << ", "
+      << folly::join(", ", aggregates) << " OVER (";
+
+  if (!partitionKeys.empty()) {
+    sql << "partition by " << folly::join(", ", partitionKeys);
+  }
+  if (!sortingKeys.empty()) {
+    sql << " order by " << folly::join(", ", sortingKeys);
+  }
+
+  sql << ") FROM tmp";
+
+  return sql.str();
+}
+
+std::optional<MaterializedRowMultiset> AggregationFuzzer::computeDuckWindow(
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<std::string>& sortingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<RowVectorPtr>& input,
+    const core::PlanNodePtr& plan) {
+  // Check if DuckDB supports specified aggregate functions.
+  auto windowNode = dynamic_cast<const core::WindowNode*>(plan.get());
+  VELOX_CHECK_NOT_NULL(windowNode);
+  for (const auto& window : windowNode->windowFunctions()) {
+    if (duckFunctionNames_.count(window.functionCall->name()) == 0) {
+      return std::nullopt;
+    }
+  }
+
+  const auto& outputType = plan->outputType();
+
+  if (!isDuckSupported(input[0]->type()) || !isDuckSupported(outputType)) {
+    return std::nullopt;
+  }
+
+  return computeDuckResults(
+      makeDuckWindowSql(
+          partitionKeys,
+          sortingKeys,
+          aggregates,
+          asRowType(input[0]->type())->names()),
+      input,
+      outputType);
+}
+
+void AggregationFuzzer::verifyWindow(
+    const std::vector<std::string>& partitionKeys,
+    const std::vector<std::string>& sortingKeys,
+    const std::vector<std::string>& aggregates,
+    const std::vector<RowVectorPtr>& input,
+    bool orderDependent) {
+  std::stringstream frame;
```

Comment:
Could you either comment or add check the size of aggregates is one that we only support one aggregation for now? Thanks!